src/averageAges.js

```diff
@@ -34,7 +43,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const allWomen = people.filter(item => item.sex === 'f');
+  const womenWithKids = allWomen.filter(item =>
+    people.some(value => value.mother === item.name));
```

Comment:
try to use one array and filter it by the right condition depend on the withChildren parameter